src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2421,33 +2552,43 @@ public override BoundNode VisitAddressOfOperator(BoundAddressOfOperator node)
             return null;
         }
 
-        private void WarnOnNullReferenceArgument(BoundExpression argument, TypeSymbolWithAnnotations argumentType, ParameterSymbol parameter, bool expanded)
+        private bool ReportNullReferenceArgumentIfNecessary(BoundExpression argument, TypeSymbolWithAnnotations argumentType, ParameterSymbol parameter, TypeSymbolWithAnnotations paramType)
```

Comment:
For people with short attention spans like me, it would be great to put some examples as comments.

`// For example, with 'void M(string x)', warn on the faulty argument in 'M(null)' or 'M(somePossiblyNullValue)'`